### PR TITLE
services/horizon: Error when setting BUCKET_DIR_PATH through --captive-core-config-path

### LINKED ISSUE #3699: BUCKET_DIR_PATH should be overwritten to stellarCoreRunner.storagePath
### What version are you using?

2.4.1

### What did you do?

If `--captive-core-config-append-path` file contains `BUCKET_DIR_PATH` it gets propagated to generated `stellar-core.cfg` file used by Captive-Core.

### What did you expect to see?

`BUCKET_DIR_PATH` is ignored and the final `BUCKET_DIR_PATH` is set to `stellarCoreRunner.storagePath`+`/buckets` (generated using `--captive-core-storage-path`). Or `BUCKET_DIR_PATH` is removed from the final config file completely (then defaults to `buckets` in working dir).

### What did you see instead?

When running multiple Stellar-Cores (ex. parallel reingestion) `BUCKET_DIR_PATH` is reused causing:
```
Got an exception: unable to flock file: /BUCKET_DIR_PATH/stellar-core.lock (Resource temporarily unavailable). This can be caused by access rights issues or another stellar-core process already running
```

### PR DESCRIPTION
**Pull request by 2opremio**
Fixes #3699

### CODE REVIEW

**Comment by tamirms** [comment]
looks good! can you also update the changelog and add a test case which exercises the error path?

**Comment by 2opremio** [comment]
Sure thing!

**Comment by 2opremio** [comment]
PTAL

**Review by tamirms — APPROVED** [review]

File: services/horizon/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ file. This project adheres to [Semantic Versioning](http://semver.org/).
 * Handle replica conflict errors gracefully ([3674](https://github.com/stellar/go/pull/3674)).
 * Fix data race in request parameters handling ([3690](https://github.com/stellar/go/pull/3690)).
 * Fix bug where the configuration for `CAPTIVE_CORE_LOG_PATH`, `CAPTIVE_CORE_PEER_PORT`, and `CAPTIVE_CORE_HTTP_PORT` were ignored if they were configured via environment variables instead of command line arguments. ([3702](https://github.com/stellar/go/pull/3702)).
+* Error when setting `BUCKET_DIR_PATH` through `--captive-core-config-path` ([3707](https://github.com/stellar/go/pull/3707)).
```
Comment (bartekn) [inline services/horizon/CHANGELOG.md:22]:
```suggestion
* Fix error when setting `BUCKET_DIR_PATH` through `--captive-core-config-path` ([3707](https://github.com/stellar/go/pull/3707)).
```

**Review by bartekn — APPROVED** [review]

File: services/horizon/CHANGELOG.md (hunk 1 above)
Comment (2opremio) [reply services/horizon/CHANGELOG.md:22]: Error is a verb here

**Pull request merged**